Pseudo PR for nf-core/scrnaseq: 1.0.0 Release (#16)
nf-core/scrnaseq: docs/usage.md

  * [`-profile`](#-profile)
  * [`--reads`](#--reads)
  * [`--singleEnd`](#--singleend)
  * [`--aligner`](#--aligner)
* [Cellular barcodes](#cellular-barcodes)
  * [`--type` to specify droplet type](#--type-to-specify-droplet-type)
  * [`--tenx_version` (using cellranger barcodes)](#--tenx_version-cellranger-barcodes)
* [Reference genomes](#reference-genomes)
  * [`--genome` (using iGenomes)](#--genome-using-igenomes)
  * [`--fasta`](#--fasta)
  * [`--gtf`](#--gtf)
  * [`--transcript_fasta`](#--transcript_fasta)
  * [`--salmon_index`](#--salmon_index)
  * [`--txp2gene`](#--txp2gene)
  * [`--kallisto_gene_map`](#--kallisto_gene_map)

>   * [`--igenomesIgnore`](#--igenomesignore)

Reviewer: Don't we have `igenomes_ignore` now?

NXF_OPTS='-Xms1g -Xmx4g'
```

## Running the pipeline

The typical command for running the pipeline is as follows:

```bash

> nextflow run nf-core/scrnaseq --reads '*_R{1,2}.fastq.gz' --type '10x' --aligner 'kallisto' --barcode_whitelist 10xv2.whitelist.txt --fasta human.fasta --gtf human.gtf -profile docker

Reviewer: Do we really need quotes around `kallisto`?
Typical command is run with Kallisto then I suppose?
Should it not be specified by default in the config files? (with the specific files)

```bash
work            # Directory containing the nextflow working files
results         # Finished results (configurable, see below)
.nextflow.log   # Log file from Nextflow
# Other nextflow hidden files, eg. history of pipeline runs and old logs.
```

### Updating the pipeline

When you run the above command, Nextflow automatically pulls the pipeline code from GitHub and stores it as a cached version. When running the pipeline after this, it will always use the cached version if available - even if the pipeline has been updated since. To make sure that you're running the latest version of the pipeline, make sure that you regularly update the cached version of the pipeline:

```bash
nextflow pull nf-core/scrnaseq
```

### Reproducibility

It's a good idea to specify a pipeline version when running the pipeline on your data. This ensures that a specific version of the pipeline code and software are used when you run your pipeline. If you keep using the same tag, you'll be running the same version of the pipeline, even if there have been changes to the code since.

> First, go to the [nf-core/scrnaseq releases page](https://github.com/nf-core/scrnaseq/releases) and find the latest version number - numeric only (eg. `1.3.1`). Then specify this when running the pipeline with `-r` (one hyphen) - eg. `-r 1.3.1`.

Reviewer: ```suggestion
First, go to the [nf-core/scrnaseq releases page](https://github.com/nf-core/scrnaseq/releases) and find the latest version number - numeric only (eg. `1.0.0`). Then specify this when running the pipeline with `-r` (one hyphen) - eg. `-r 1.0.0`.
```

Reviewer: Should be good to mention a release that we actually have, we should have a TODO for that in the template I think.

Author: First one will be 1.0.0 - I adjusted this accordingly now :-)

It is not possible to run a mixture of single-end and paired-end files in one run.

### `--aligner`

The workflow can handle three types of methods:

* Kallisto/Bustools
* Salmon Alevin + AlevinQC
* STARsolo

> To choose which one to use, please specify either `alevin`, `star` or `kallisto` as a parameter option for `--aligner`.

Reviewer: Is there a default one?

}
```

### `--fasta`

If you prefer, you can specify the full path to your reference genome when you run the pipeline:

```bash
--fasta '[path to Fasta reference]'
```

> > Note that you need to specify either a `--genome` or `--fasta` when running the STARsolo workflow. The Kallisto and Alevin workflows can utilize a `--transcript_fasta` instead, whereas STAR needs a genomic FastA file as input in all cases.

Reviewer: ```suggestion
> Note that you need to specify either a `--genome` or `--fasta` when running the STARsolo workflow. The Kallisto and Alevin workflows can utilize a `--transcript_fasta` instead, whereas STAR needs a genomic fasta file as input in all cases.
```

Reviewer: isn't it fasta files? or is it actually FastA files? Have I been lied to my whole life?

### `--gtf`

Specify a valid GTF file for the workflow here.

### `--transcript_fasta`

> If you intend to skip the generation of a transcriptomic FastA file, you can use this parameter to supply a transcriptomic FastA file here. If you don't specify this, it will be automatically generated from the supplied genomics FastA file utilizing the GTF annotation subsequently.

Reviewer: same question here.
`fasta` or `FastA`?

### Custom resource requests

Wherever process-specific requirements are set in the pipeline, the default value can be changed by creating a custom config file. See the files hosted at [`nf-core/configs`](https://github.com/nf-core/configs/tree/master/conf) for examples.

If you are likely to be running `nf-core` pipelines regularly it may be a good idea to request that your custom config file is uploaded to the `nf-core/configs` git repository. Before you do this please can you test that the config file works with your pipeline of choice using the `-c` parameter (see definition below). You can then create a pull request to the `nf-core/configs` repository with the addition of your config file, associated documentation file (see examples in [`nf-core/configs/docs`](https://github.com/nf-core/configs/tree/master/docs)), and amending [`nfcore_custom.config`](https://github.com/nf-core/configs/blob/master/nfcore_custom.config) to include your custom profile.

> If you have any questions or issues please send us a message on [Slack](https://nf-core-invite.herokuapp.com/).

Reviewer: ```suggestion
If you have any questions or issues please send us a message on [Slack](https://nfcore.slack.com/channels/scrnaseq/).
```
